data/wp/wp-content/mu-plugins/EPFL-SC-people.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * Plugin Name: EPFL People shortcode
+ * Plugin URI: https://github.com/epfl-idevelop/EPFL-WP-SC-People
+ * Description: provides a shortcode to display results from People
+ * Version: 1.0
+ * Author: Emmanuel JAEP
+ * Author URI: https://people.epfl.ch/emmanuel.jaep?lang=en
+ * License: Copyright (c) 2017 Ecole Polytechnique Federale de Lausanne, Switzerland
+ **/
+
+function epfl_people_log( $message ) {
+    if ( WP_DEBUG === true ) {
+        if ( is_array( $message ) || is_object( $message ) ) {
+            error_log( print_r( $message, true ) );
+        } else {
+            error_log( $message );
+        }
+    }
+}
+
+function epfl_people_urlExists( $url )
+{
+    $handle = curl_init( $url );
+    curl_setopt( $handle, CURLOPT_RETURNTRANSFER, TRUE );
+
+    $response = curl_exec( $handle );
+    $httpCode = curl_getinfo( $handle, CURLINFO_HTTP_CODE );
+
+    if ( $httpCode >= 200 && $httpCode <= 400 ) {
+        return true;
+    } else {
+        return false;
+    }
+    curl_close( $handle );
+}
+
+function epfl_people_process_shortcode( $attributes, $content = null )
+{
+    $attributes = shortcode_atts( array(
+        'url' => ''
+    ), $attributes );
+
+    // Sanitize parameter
+    $url = sanitize_text_field( $attributes['url'] );
+
+    // Check if the result is already in cache
+    $result = wp_cache_get( $url, 'epfl_people' );
+    if ( false === $result ){
```

Comment:
Il te manque un bloc `else` pour traiter le cas dans lequel tu aurais l'info dans le cache 😉

Reply:
En effet, il manquait un bloc. J'ai corrigé.

Ce qui faut comprendre avec le cache de WP c'est qu'il sera utilisé seulement si dans la même page on aura plusieurs fois la même URL de people qui est appelé. (Et j'ai testé, si je mets 2 fois le même shortcode people dans la même page, le cache est utilisé)

La raison :
"By default, the object cache is non-persistent. This means that data stored in the cache resides in memory only and only for the duration of the request. Cached data will not be stored persistently across page loads unless you install a persistent caching plugin." 

extrait de la doc : https://codex.wordpress.org/Class_Reference/WP_Object_Cache#Persistent_Caching

Donc il me semble que le cache ne sera pas utilisé souvent :-)